File: src/main/java/uk/edu/populationfitness/models/genes/cache/SelfIdentifiedGenes.java
package uk.edu.populationfitness.models.genes.cache;

import uk.edu.populationfitness.models.genes.GenesIdentifier;

/**
 * A bit set identifier that directly references the bitset.
 */
public class SelfIdentifiedGenes implements GenesIdentifier {
    public final long[] integerGenes;

    public SelfIdentifiedGenes(long[] integerGenes) {
        this.integerGenes = integerGenes;
    }

    @Override
    public long asUniqueLong() {
        throw new Error("Cannot convert to long");
    }
}

File: DotNet/PopulationFitness/PopulationFitness/Models/Genes/Cache/SelfIdentifiedGenes.cs
using System;

namespace PopulationFitness.Models.Genes.Cache
{

    /**
     * A bit set identifier that directly references the bitset.
     */
    public class SelfIdentifiedGenes : IGenesIdentifier
    {
        public readonly long[] integerGenes;

        public SelfIdentifiedGenes(long[] integerGenes)
        {
            this.integerGenes = integerGenes;
        }

        public long AsUniqueLong()
        {
            throw new Exception("Cannot convert to long");
        }
    }
}
